wvd-templates/DSC/Functions.ps1

```diff
@@ -401,4 +294,28 @@ function ImportRDPSMod {
 function GetCurrSessionHostName {
     $Wmi = (Get-WmiObject win32_computersystem)
     return "$($Wmi.DNSHostName).$($Wmi.Domain)"
+}
+
+function GetSessionHostDesiredStates {
+    return ('Available', 'NeedsAssistance')
+}
+
+function IsRDAgentRegistryValidForRegistration {

```

Comment:
Consider having this return null if successful but otherwise return the error message so that the calling code, such as  Script-RegisterSessionHost.ps1  can throw or log the exact error message.